Modules/TRD/src/TrackletCountCheck.cxx

```diff
@@ -196,4 +212,91 @@ void TrackletCountCheck::beautify(std::shared_ptr<MonitorObject> mo, Quality che
   }
 }
 
+void TrackletCountCheck::reset()
+{
+  ILOG(Debug, Devel) << "TrackletCountCheck::reset" << ENDM;
+}
+
+void TrackletCountCheck::startOfActivity(const Activity& activity)
+{
+  ILOG(Debug, Devel) << "TrackletCountCheck::start : " << activity.mId << ENDM;
+  mActivity = make_shared<Activity>(activity);
+
+  ILOG(Debug, Devel) << "initializing TrackletCountCheck" << ENDM;
+
+  mThresholdMeanHighPerTimeFrame = std::stof(mCustomParameters.atOrDefaultValue("UpperthresholdPerTimeFrame", "600000.0" /*default value*/, *mActivity));
```

Comment:
I stumbled upon this as well and don't really like the framework API. Basically you should not use the `atOrDefaultValue` method, as this will give you always the default value from the code and not from the json unless you have provided a value for the explicit activity `type` and `beamType`. Took me also a bit to figure out what to use. Please change all these to use `atOptional` like below:
```suggestion
  mThresholdMeanHighPerTimeFrame = std::stof(mCustomParameters.atOptional("UpperthresholdPerTimeFrame", *mActivity).value_or("6e5"));
```
I also think that with the scientific notation long numbers are more easy to read..